pkgs/stdenv/adapters.nix

```
@@ -238,4 +238,23 @@ rec {
         allowSubstitutes = false;
       });
     });
+
```

While we are at it maybe we should add something like these:
```suggestion
withStdenv = newStdenv: pkg:
    pkg.override { stdenv = newStdenv; };
    
withSrc = pkg: src: 
    pkg.overrideAttrs (_: { inherit src; });
```

---

In another PR. This looks great.